security/pkg/nodeagent/vm/nodeagent_v2.go

```diff
@@ -0,0 +1,136 @@
+	caClient caClientInterface.Client
+	identity string
+	certUtil util.CertUtil
+}
+
+// Start starts the node Agent.
+func (na *nodeAgentInternalV2) Start() error {
+	if na.config == nil {
+		return fmt.Errorf("node Agent configuration is nil")
+	}
+
+	if !na.pc.IsProperPlatform() {
+		return fmt.Errorf("node Agent is not running on the right platform")
+	}
+
+	log.Infof("Node Agent V2 starts successfully.")
+
+	retries := 0
+	retrialInterval := na.config.CAClientConfig.CSRInitialRetrialInterval
+	identity, err := na.pc.GetServiceIdentity()
+	if err != nil {
+		return err
+	}
+	na.identity = identity
+	var success bool
+	for {
+		log.Infof("Sending CSR (retrial #%d) ...", retries)
+		keyPEM, certChainPEM, csrError := na.sendCSRUsingCANewProtocol()
+
+		if csrError != nil {
+			log.Errorf("%v", csrError)
+			success = false
+		} else {
+			waitTime, ttlErr := na.certUtil.GetWaitTime([]byte(certChainPEM[0]), time.Now())
+			if ttlErr != nil {
+				log.Errorf("Error getting TTL from approved cert: %v", ttlErr)
+				success = false
+			} else {
+				var certChain []byte
+				for _, c := range certChainPEM {
+					certChain = append(certChain, []byte(c)...)
+				}
+				if err = caclient.SaveKeyCert(na.config.CAClientConfig.KeyFile,
+					na.config.CAClientConfig.CertChainFile,
+					keyPEM, certChain); err != nil {
+					return err
+				}
+				log.Infof("CSR is approved successfully. Will renew cert in %s", waitTime.String())
+				fmt.Println(certChainPEM)
+				retries = 0
+				retrialInterval = na.config.CAClientConfig.CSRInitialRetrialInterval
+				timer := time.NewTimer(waitTime)
+				<-timer.C
+				success = true
+			}
+		}
+
+		if !success {
+			if retries >= na.config.CAClientConfig.CSRMaxRetries {
+				return fmt.Errorf(
+					"node agent can't get the CSR approved from Istio CA after max number of retries (%d)",
+					na.config.CAClientConfig.CSRMaxRetries)
+			}
+			retries++
+			timer := time.NewTimer(retrialInterval)
+			// Exponentially increase the backoff time.
+			retrialInterval *= 2
+			<-timer.C
+		}
+	}
+}
+
+func (na *nodeAgentInternalV2) sendCSRUsingCANewProtocol() ([]byte, []string, error) {
+	options := pkiutil.CertOptions{
+		Host:       na.identity,
+		Org:        na.config.CAClientConfig.Org,
+		RSAKeySize: na.config.CAClientConfig.RSAKeySize,
+		IsDualUse:  na.config.DualUse,
+	}
+
+	csrPEM, keyPEM, err := pkiutil.GenCSR(options)
+	if err != nil {
+		return nil, nil, fmt.Errorf("failed to generated key cert for %q: %v", na.identity, err)
+	}
+	jwt, err := na.pc.GetAgentCredential()
+	if err != nil {
+		return nil, nil, fmt.Errorf("error getting JWT from metadata server for %q: %v", na.identity, err)
+	}
+	stsClient := stsclient.NewPlugin()
+	// TODO(pitlv2109): Periodically update the JWT and OAuth2 token.
+	// TODO(pitlv2109): Extract the project_id field from the JWT.
+	token, _, _ := stsClient.ExchangeToken(context.Background(), "csm-fsa@mesh-expansion-235023.iam.gserviceaccount.com", string(jwt))
```

Comment:
even this is a collab branch, checking to source code of a hardcoded project is not a good idea. either leave as a env var/config or even an invalid string is okay.

Reply:
I agree. This is a WIP PR. Once the structure of this PR looks good to you, I will implement the rest of the TODOs.